### DISCUSSION TITLE
Distinguer l'identité des élèves et leurs parcours par année scolaire

### DISCUSSION THREAD

**Discussion by ThierryM** [DSMejantel/Ecole_inclusive#11] [Ideas]
Re-Adieu David,
Afin de permettre un suivi pluri-annuel des élèves, il me semble qu'il serait intéressant de scinder la table "eleve" en 2 parties :
- la table 'eleve' ne contiendrait que les données d'identité (données en principe immuables ou pérennes): ine, nom, prénom, sexe, date-naissance, adresse, CP, commune, commentaire
- une table 'parcours' qui contiendrait pour une année scolaire donnée : annee, etablissement, classe, référent, niveau, ine (lié à la table 'eleve')
- une table 'annee_scolaire' qui listerait les années scolaires.

Qu'en penses-tu ?
A lèu.

**Comment by DSMejantel** [comment]
En effet, cela peut être utile d'avoir ce suivi sur le long terme. 
On peut rajouter à ta proposition l'identité de l'AESH.
Je vois bien une alerte automatique entre juin et septembre sur la page d'accueil de l'admin et des coordo pour préparer le changement d'année.
Il faut revoir mes derniers modules d'import et de mise à jour pour ne changer que la table parcours.

**Comment by DSMejantel** [comment]
Adieu,
École Inclusive gère désormais le suivi pluri-annuel avec une table qui enregistre les infos de l'année en cours. Les données AESH et référents sont finalement stockées en texte afin de garder une trace même sir ces personnes ne sont plus présentes sur la base.
L'INE et les données annexes sont gérées également dans la fiche élève.
Cet historique est alimenté et raffraîchi à chaque passage sur notification.sql
Depuis l'espace 'paramètres', dans l'onglet 'comptes', il est possible de le générer pour les élèves déjà présent dans la base.

Il reste à revoir les format d'importation des élèves en masse depuis SIECLE ou ONDE.
Quel est le schéma des colonnes issues d'ONDE ? Tu m'as parlé d'un problème avec le format date de naissance sur tes tests.

**Comment by ThierryM** [comment]
Adieu David,
Impeccable : je vais tester tout ça !
Je te joins une extraction.csv de ONDE issu d'un bac à sable (pas de données réelles). Le format de la date de naissance est bonne d'origine, a priori : YYYY-MM-DD. Mes soucis devaient venir de mes manipulations dans LibreOffice Calc qui ont dû modifier le format.
A lèu
[CSVExtraction_bac_à_sable.csv](https://github.com/DSMejantel/Ecole_inclusive/files/15366000/CSVExtraction_bac_a_sable.csv)

**Comment by ThierryM** [comment]
Adieu David,
Je viens de mettre à jour la dernière version de SQLpage 0.21 (pour ne pas avoir l'erreur apparue avec la version 0.20.5 et les bases SQLite) et de récupérer la dernière version de École Inclusive.
Tout à l'air de fonctionner correctement (non testé de façon très approfondie cependant) : impeccable. Mercé plan !
Seul souci avec l'import de compte élève à partir de ONDE mais c'est normal sans adaptation spécifique.
Pour info, j'ai créé sur mon site [une page](https://lofurol.fr/joomla/logiciels-libres/104-bases-de-donnees/356-sqlpage-commandes-utiles-astuces) sur des "astuces" facilitant l'utilisation de SQLPage : personnalisation du footer, pré-remplissage de tables ou travail avec des dates.
Bona seguida de dimenjada longa.

**Comment by DSMejantel** [comment]
Adieu,
En effet, j'ai testé et cela fonctionne correctement. Cela génère de nombreux logs.
Je viens de simplifier avec la plupart des ::int qui ne servait à rien !
POur certains il a fallu reécrier avec le CAST as integer.
Cela m'a permis une relecture qui a permis de repérer deux erreurs d'appel à des icônes et d'améliorer une page d'édition des comptes.
Je mets sur Github après l'orage !
A lèu

**Reply by lovasoa** [reply]
Bonjour David! Est-ce que la nouvelle version a des logs plus verbeux que ce qu'il faudrait pour toi?

**Comment by DSMejantel** [comment]
Bonjour à vous deux,
Rien à redire sur le travail des log. Dans le terminal, de nombreuses lignes apparaissaient pour signifier l'avertissement sur l'écriture non adaptée en sqlite mais portée tout de même par SQLpage.
Désormais tous les fichiers sont corrigés sur le dépôt et celame semble plus propre.
Je me rappelle avoir été fier de ma découverte de ::int au début de ma programmation et de l'utilisation de variable. Cette portabilité de SQLpage pour ma base sqlite m'a finalement permis de dépasser un écueil qui m'aurait peut-être empêché de pousser plus loin mes limites !
